iceoryx_posh/CMakeLists.txt

```diff
@@ -209,11 +201,10 @@ if(TOML_CONFIG)
     ######### posh config lib ##########
     #
     iox_add_library(
-        NO_EXPORT
```

Comment:
@elBoberido By removing `NO_EXPORT` the posh header files are now installed five times for each packages. It was like this before. I'm not sure what is the idiomatic CMake way here. I would expect the installation only for `PUBLIC_LIBS`. However, neither making them public nor removing `INSTALL_INTERFACE` prevented the header being installed for five times.

Reply:
I think we need some reorganization of the folder structure and also of this CMakeLists.txt. It was not good to have that many libs in the same CMakeLists.txt nor to have the header of these libs in the same folder structure. For better separation it might be necessary to have something like this
```
iceoryx_posh
|- config (I'm not quite sure why this exists)
|  |- include
|  |  |- iceoryx_posh
|  |     |- config
|  |        |- foo.hpp
|  |- source
|     |- foo.cpp
|  |- CMakeLists.txt
|- posh
|  |- include
|  |  |- iceoryx_posh
|  |     |- capro
|  |     |  |- service_description.hpp
|  |     |- mepoo
|  |- source
|  |- CMakeLists.txt
|- roudi
|  |- include
|- testing
   |- include
```

Or even go further and have separate folder structures for `capro`, `mepoo`, etc. With an own CMakeLists.txt. These would create object libs and the main CMakelists.txt would stick them together to binary libs. Something similar could be done in iceoryx_hoofs to better separate the modules from each other and prevent circular dependencies.

@dkroenke what do you think?

Reply:
I like the idea, but this is something for a follow-up PR. See [this comment](https://github.com/eclipse-iceoryx/iceoryx/issues/590#issuecomment-1151422780).